### ISSUE TITLE
Access the original response in resolver

### ISSUE THREAD

**Issue by ArielGueta** [mswjs/interceptors#266]
Thanks for this library.

I'm wondering if there is any way to intercept the original response in the resolve phase:

```ts
createInterceptor({
    modules: [interceptXMLHttpRequest],
    async resolver(req, ref) {
       const originalResponse = await ??
       return modifyResponse(originalResponse)
   }
)
```

Thanks.

**Comment by kettanaito** [comment]
Hey, @ArielGueta. 

You can listen for any kind of response using the "response" event of the interceptor:

```js
const interceptor = createInterceptor(...)

interceptor.on('response', (request, response) => {
  console.log(response)
})
```

https://github.com/mswjs/interceptors/blob/ca5cc325ef935d610ee33657e76e94e95420c500/src/createInterceptor.ts#L45-L48

This library is not concerned about whether the response was mocked or not, because it's not concerned with mocking in the first place. You can intercept requests and mock responses, but then it becomes your responsibility to distinguish between original and mocked responses. This is something we do in MSW by keeping track of `request.id`. 

I also see that you're using a rather old version of this library in your code example. Please make sure to update to the latest one to have all the recent fixes and improvements. Thanks.

**Comment by kettanaito** [comment]
Following up on your question, I've missed the part where you wanted to perform the original request within the resolver. You can do that but you'd have to construct the request from scratch based on the info exposed in `request`. 

Here's an example of response modification using `node-fetch`:

```js
import fetch from 'node-fetch'

createInterceptor({
    modules: [interceptXMLHttpRequest],
    async resolver(req, ref) {
       const originalResponse = await fetch(req.url.href, {
         method: req.methof,
         headers: req.headers,
         body: req.body
       })
       const json = await originalResponse.json()

       return modifyResponse(json)
   }
)

```

**Comment by ArielGueta** [comment]
But this will create an infinite loop, no?

**Comment by kettanaito** [comment]
A good catch. That's why when we do a similar thing we append a special header to the original request, forcing the interceptor to bypass requests with that header. You can check that header's presence/value in the resolver, and if it's there, do an early `return`, which would bypass such a request.

**Comment by ArielGueta** [comment]
Thanks.

**Comment by kettanaito** [comment]
## Released: v0.19.3 🎉

This has been released in v0.19.3!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.19.3)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.19.3)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
